### DISCUSSION TITLE
Enhancement request: "Send report now" button

### DISCUSSION THREAD

**Discussion by zell-mbc** [plausible/analytics#4715] [Self-Hosted Support] [answered]
I know this came up in the past already, but I'd like to ask for a "send email report now" feature again.

I am aware of the workaround to send a password reset email to test if the email setup is working in general, but I am dealing with a case where the weekly reports get stuck in my spam filter and troubleshooting means change a spam filter setting and wait a week to see if the change works.

**Comment by ruslandoga** [comment]
👋 @zell-mbc 

Until this is implemented you can trigger a report manually with something like this:

```console
$ cd plausible-ce
$ docker compose exec plausible bin/plausible remote
```
```elixir
iex> domain = "dummy.site"
iex> site = Plausible.Repo.get_by!(Plausible.Site, domain: domain)
iex> Plausible.Workers.SendEmailReport.perform(%Oban.Job{args: %{"interval" => "weekly", "site_id" => site.id}})
```

**Reply by zell-mbc** [reply]
Cool, almost there I think.
I set `domain` to one of mine, ran the Repo.get_by command which looks like it worked, but then SendEmailReport gives me an error.
When checking on my mail server it looks like it's not properly connecting to the email server.

```
postfix/submission/smtpd[51156]: connect from ip-078-
postfix/submission/smtpd[51156]: warning: non-SMTP command from ip-078-…\b\t\b\006\b\005\b\004\006\001\005\001\004\001\006\003\005\003\004\003\025\003\003\000\002\002
postfix/submission/smtpd[51156]: disconnect from ip-078-```
```

It' not even getting to the SPAM filter, maybe the email client somehow needs to be initialized/configured before running SendEmailReport?

**Reply by ruslandoga** [reply]
Please share your SMTP configuration both for Plausible and your SMTP server.

**Reply by ruslandoga** [reply]
And also the return value from the "send report" command.

**Reply by zell-mbc** [reply]
Here's the send report return value:
```
iex(plausible@e3b9759feff6)3> Plausible.Workers.SendEmailReport.perform(%Oban.Job{args: %{"interval" => "weekly", "site_id" => site.id}})
:ok

12:06:23.764 [error] Failed to send e-mail
```

Will look up the smtp settings next. The thing is, the emails sent on Monday were apparently ok, they were accepted by postfix and it was RSPAMD which sorted them out as spam.
![image](https://github.com/user-attachments/assets/7e144ca6-fdab-46b2-b5a3-c83e5f85b15d)
Sender and receiver are all known and correct.

The ones I am trying to send manually now don't get that far.

**Reply by zell-mbc** [reply]
Docker compose:
```
environment:
      - MAILER_EMAIL=service@mydomain.com
      - MAILER_NAME=service@mydomain.com
      - SMTP_HOST_ADDR=mail.mydomain.com
      - SMTP_HOST_PORT=587
      - SMTP_USER_NAME=myusername@mydomain.com
      - SMTP_USER_PWD=Passwd
      - SMTP_HOST_SSL_ENABLED=true
      - SMTP_RETRIES=2
      - LOG_LEVEL=warning
      - CRON_ENABLED=true  
      - MAILER_ADAPTER=Bamboo.Mua
```

**Reply by ruslandoga** [reply]
Can you please try either changing the port to 465 or removing SMTP_HOST_SSL_ENABLED? SMTP via port 587 usually uses STARTTLS. And 465 is for SMTPS (SMTP_HOST_SSL_ENABLED=true).

**Reply by zell-mbc** [reply]
Sorry, took a while to get back to you.
I whitelisted the Plausible server on my email server and this morning the email reports came through, with SSL_ENABLED still on. All reports show zero traffic even though the Plausible website shows traffic but that's another issue I think.

![image](https://github.com/user-attachments/assets/254776e9-60ee-497a-8153-9f508e865a82)

As you suggested I went ahead just now and and removed the SMTP_HOSTS_SSL_ENABLED setting, restarted the docker-compose  and tried the above `plausible remote`  commands.

Still the same "Failed to send Email" error. So somehow the automated routine does something different than what I do manually.

**Reply by ruslandoga** [reply]
You might be running two Plausible instances :) Or at least it looks like that a bit: the report might come from the one that doesn't get events but has SMTP configured "correctly" somehow, and the other one might be getting events but has SMTP problems.

**Reply by ruslandoga** [reply]
Would you be able to show your SMTP-related configuration (the way Plausible sees it) and try sending a test email somewhere:
```console
$ cd plausible-ce
$ docker compose exec plausible bin/plausible remote
```
```elixir
iex> Application.get_env(:plausible, Plausible.Mailer)
#==> ... what does it return? ...

# pick one of your own mailboxes
iex> receiver = "your-username-here@gmail.com"
iex> email = Bamboo.Email.new_email(to: receiver, from: PlausibleWeb.Email.mailer_email_from(), text_body: "test")
iex> Plausible.Mailer.deliver_now(email)
```

**Reply by zell-mbc** [reply, marked as answer]
> You might be running two Plausible instances :) 

You got me worried for a moment, because I did indeed move things around before settling down on the current server and that DNS might point at a leftover instance isn't totally unrealistic, but no, did a quick `docker compose down` and both my `docker exec` shell I used to collect the information below, as well as the website which shows active traffic are gone. That would have been an embarrassing one :-)

Back to our subject:

When running the show mailer config command I noticed that SSL was still on, looks like a compose restart didn't do the trick. After a compose down / up I get ` ssl: [middlebox_comp_mode: false]`and the email test works!

As well as manually sending a report, and this one even has got data in it?!
Left sent manually just now, right what was sent tonight
![image](https://github.com/user-attachments/assets/ea7b8743-1c33-4d1b-8ea9-351f7f6b25e9)

I'll think about this some more and will check the logs, but the "send-report-now" case is solved! Thanks for helping along…

**Reply by ruslandoga** [reply]
> but no, did a quick docker compose down and both my docker exec shell I used to collect the information below, as well as the website which shows active traffic are gone

That would be the second, active instance in my example.
